packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -2183,14 +2183,20 @@ class _InputDecoratorState extends State<InputDecorator> with TickerProviderStat
         .merge(decoration!.floatingLabelStyle ?? decoration!.labelStyle);
     }
 
-    final TextStyle? style = MaterialStateProperty.resolveAs(decoration!.floatingLabelStyle, materialState)
+    final TextStyle? floatingLabelStyle = MaterialStateProperty.resolveAs(decoration!.floatingLabelStyle, materialState)
```

Comment:
It should be possible to override the floatingLabelStyle for `MaterialState.error`. Perhaps what we need here is an example.